filebeat/input/journald/input.go

```diff
@@ -204,3 +205,44 @@ func seekBy(log *logp.Logger, cp checkpoint, seek, defaultSeek journalread.SeekM
 	}
 	return mode, cp.Position
 }
+
+// readerAdapter is an adapter so journalread.Reader can
+// behave like reader.Reader
+type readerAdapter struct {
+	r        *journalread.Reader
+	canceler input.Canceler
+}
+
+func (r *readerAdapter) Close() error {
+	return r.r.Close()
+}
+
+func (r *readerAdapter) Next() (reader.Message, error) {
+	data, err := r.r.Next(r.canceler)
+	if err != nil {
+		return reader.Message{}, err
+	}
+
+	content := []byte(data.Fields["MESSAGE"])
+	delete(data.Fields, "MESSAGE")
```

Comment:
```suggestion
	err = data.Fields.Delete("MESSAGE")
	if err != nil {
	    return reader.Message{}, fmt.Errorf("error deleting MESSAGE key: %+v", err)
	}
```

Reply:
Unfortunately, that won't work, `data.Fields` is a `map[string]string`, hence it hasn't got a `Delete` method. `data` is a `sdjournal.JournalEntry` (from `github.com/coreos/go-systemd/v22/sdjournal`).

Reply:
Oh yes, my bad. I thought it `r.r.Next` returns `reader.Message` not a `journalread.Entry`. :facepalm: